src/cli/utils.ts

```diff
@@ -168,37 +170,52 @@ const tokenToDevGatewayMap: Record<TokenType, string> = {
   // matic: 'https://rpc-amoy.polygon.technology',
 };
 
-export function configFromOptions({
-  gateway,
-  dev,
-  token,
-}: GlobalOptions): TurboUnauthenticatedConfiguration {
+export function configFromOptions(
+  options: GlobalOptions,
+): TurboUnauthenticatedConfiguration {
   let config: TurboUnauthenticatedConfiguration = {};
 
-  if (dev) {
+  const token = tokenFromOptions(options);
+  config.token = token;
+
+  if (options.dev) {
     config = developmentTurboConfiguration;
     config.gatewayUrl = tokenToDevGatewayMap[token];
   } else {
     config = defaultTurboConfiguration;
   }
 
   // If gateway is provided, override the default or dev gateway
-  if (gateway !== undefined) {
-    config.gatewayUrl = gateway;
+  if (options.gateway !== undefined) {
+    config.gatewayUrl = options.gateway;
   }
 
-  config.token = token;
-
   return config;
 }
 
+export async function turboFromOptions(
+  options: WalletOptions,
+): Promise<TurboAuthenticatedClient> {
+  const privateKey = await privateKeyFromOptions(options);
+
+  return TurboFactory.authenticated({
+    ...configFromOptions(options),
+    privateKey,
+  });
+}
+
 export function getUploadFolderOptions(options: UploadFolderOptions): {
   folderPath: string;
   indexFile: string | undefined;
   fallbackFile: string | undefined;
   disableManifest: boolean;
   maxConcurrentUploads: number;
 } {
+  console.log('options', options);
```

Comment:
nit: looks like this was used for debugging

Reply:
nice catch c3ad1e7